src/OpenTelemetry.Collector.Dependencies/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -0,0 +1,126 @@
+﻿// <copyright file="SqlClientDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Collector.Dependencies.Implementation
+{
+    internal class SqlClientDiagnosticListener : ListenerHandler
+    {
+        internal const string SqlDataBeforeExecuteCommand = "System.Data.SqlClient.WriteCommandBefore";
+        internal const string SqlMicrosoftBeforeExecuteCommand = "Microsoft.Data.SqlClient.WriteCommandBefore";
+
+        internal const string SqlDataAfterExecuteCommand = "System.Data.SqlClient.WriteCommandAfter";
+        internal const string SqlMicrosoftAfterExecuteCommand = "Microsoft.Data.SqlClient.WriteCommandAfter";
+
+        internal const string SqlDataWriteCommandError = "System.Data.SqlClient.WriteCommandError";
+        internal const string SqlMicrosoftWriteCommandError = "Microsoft.Data.SqlClient.WriteCommandError";
+
+        private readonly PropertyFetcher commandFetcher = new PropertyFetcher("Command");
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher dataSourceFetcher = new PropertyFetcher("DataSource");
+        private readonly PropertyFetcher databaseFetcher = new PropertyFetcher("Database");
+        private readonly PropertyFetcher commandTextFetcher = new PropertyFetcher("CommandText");
+        private readonly PropertyFetcher exceptionFetcher = new PropertyFetcher("Exception");
+
+        public SqlClientDiagnosticListener(string sourceName, Tracer tracer)
+            : base(sourceName, tracer)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+        }
+
+        public override void OnCustom(string name, Activity activity, object payload)
+        {
+            switch (name)
+            {
+                case SqlDataBeforeExecuteCommand:
+                case SqlMicrosoftBeforeExecuteCommand:
+                    {
+                        var command = this.commandFetcher.Fetch(payload);
+
+                        if (command == null)
+                        {
+                            CollectorEventSource.Log.NullPayload(nameof(SqlClientDiagnosticListener) + name);
```

Comment:
Need separator between `SqlClientDiagnosticListener` and `name`. Current structure would look like `SqlClientDiagnosticListenerSystem.Data.SqlClient.WriteCommandBefore`, the suggestion output would be `SqlClientDiagnosticListenerSystem-System.Data.SqlClient.WriteCommandBefore`
```suggestion
                            CollectorEventSource.Log.NullPayload($"{nameof(SqlClientDiagnosticListener)}-{name}");
```